### ISSUE TITLE
TF 0.15: Error ConflictsWith access_token conflicts with credentials

### ISSUE THREAD

**Issue by SkYNewZ** [hashicorp/terraform-provider-google#8930]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

On Terraform Cloud

```
Terraform v0.15.0
on linux_amd64
```

### Affected Resource(s)

* `data.google_service_account_access_token`

### Terraform Configuration Files

```tf
# main.tf
data "google_service_account_access_token" "default" {
  provider               = google.superuser
  target_service_account = null_resource.wait_group_iam.triggers.sa_email
  scopes                 = ["userinfo-email", "cloud-platform"]
  lifetime               = "1200s"
}

# provider.tf
provider "google" {
  alias                 = "superuser"
  project               = var.project_id
  request_timeout       = "120s"
  user_project_override = true
}

# versions.tf
terraform {
  required_version = ">= 0.14"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "3.64" # Same issue with previous used version 3.48
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "3.64" # Same issue with previous used version 3.48
    }
  }
}
```

### Debug Output

* https://gist.github.com/SkYNewZ/59da86053233b4e24b445c3492fe8293#file-plan-log
* Highlights to
```
2021-04-15T08:41:34.061Z [WARN]  Provider "provider[\"registry.terraform.io/hashicorp/google\"].superuser" produced an unexpected new value for data.google_service_account_access_token.default.
      - .access_token: inconsistent values for sensitive attribute
```

### Expected Behavior

Terraform stuff as usual. Here a successful destroy plan

### Actual Behavior

```
╷
│ Error: ConflictsWith
│ 
│ "access_token": conflicts with credentials
╵
╷
│ Error: ConflictsWith
│ 
│ "access_token": conflicts with credentials
╵
```

### Steps to Reproduce

1. To have a classic workspace, correctly deployed, using Terraform v0.14.X
2. Make some changes to be able to use Terraform 0.15 (deprecated functions and other stuff)
3. Use Terraform 0.15 binary
4. Plan (apply or destroy) the workspace and this will print this error

### Important Factoids

I did not have this error in Terraform v0.14 since the datasources were not refreshed during destroy plans (https://github.com/hashicorp/terraform/pull/27408) in this version. This is why it was necessary for me to migrate to Terraform 0.15. Once this is done and on the first attempt at destruction, this is the error I get.

You will find the actual and latest version of terraform state here: https://gist.github.com/SkYNewZ/59da86053233b4e24b445c3492fe8293#file-terraform-tfstate-json

b/303616064

**Comment by upodroid** [comment]
Hi

May I ask why you are using `access_token`? If it is for service account impersonation, you can use `impersonate_service_account` field in the provider block to impersonate a service account directly without having to use a `google_service_account_access_token` datasoruce. It also refreshes the access token when it expires instead of the `google_service_account_access_token` datasource which will break any terraform apply operation that takes longer than the lifetime of the token.

**Comment by benhxy** [comment]
Are you still seeing this error? 

According to https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/service_account_access_token.html, the "provider" field should be set to google, not google.superuser.

**Comment by rileykarson** [comment]
> the "provider" field should be set to google, not google.superuser.

That's a Terraform provider alias: https://developer.hashicorp.com/terraform/language/providers/configuration#alias-multiple-provider-configurations

The datasource is working correctly here, and the issue is with the provider block. Retagging.

I'm going to close this as obsolete- I don't think there's enough information to move the issue forward, it's a few years old, and HashiCorp heavily discourages deferred values in providers nowadays.
